File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int res = Integer.MAX_VALUE;
        int start = 0;
        int cur = 0;
        for (int end = 0; end < nums.length; end++) {
            cur += nums[end];
            while (target <= cur) {
                res = Math.min(res, end - start + 1);
                cur -= nums[start];
                start++;
            }
        }
        if (res == Integer.MAX_VALUE) {
            return 0;
        }
        return res;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        res = float('inf')
        cur = 0
        start = 0
        for end in range(len(nums)):
            cur += nums[end]
            while cur >= target:
                res = min(res, end - start + 1)
                cur -= nums[start]
                start += 1
        if res == float('inf'):
            return 0
        return res